### PR TITLE
Hash the entire client hello (except the cookie) for hello verify

### PR DESCRIPTION
**Pull request by randombit**
This makes it simpler to statelessly verify a DTLS cookie without
having to fully parse the initial client hello. (GH #2320)

### CODE REVIEW

File: src/lib/tls/msg_client_hello.cpp
Comment (MartinOsterta) [inline src/lib/tls/msg_client_hello.cpp:288]: why not merge this if with the if in line 294 and the if in line 297?

**Review by MartinOsterta — CHANGES_REQUESTED** [review]

File: src/lib/tls/msg_client_hello.cpp
Comment (randombit) [reply src/lib/tls/msg_client_hello.cpp:270]: Thanks that approach is much clearer I do not remember why I wrote it the way I did.

**Pull request merged**
